### PR TITLE
Add FailFast for SEH handling

### PR DESCRIPTION
**Pull request by MikeHolman**
Doing SEH over Chakra frames can cause us to go into a bad state because C++ destructors will not be called.
This change adds code to explicitly FailFast in case a host tries this.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/805)

### CODE REVIEW

**Comment by MikeHolman** [comment]
@Yongqu @LouisLaf, can you please review?

File: bin/ch/ch.cpp [hunk 1]
```diff
@@ -109,9 +109,14 @@ int HostExceptionFilter(int exceptionCode, _EXCEPTION_POINTERS *ep)
...
 
     fwprintf(stderr, _u("FATAL ERROR: %ls failed due to exception code %x\n"), hostName, exceptionCode);
-    fflush(stderr);
 
-    return EXCEPTION_EXECUTE_HANDLER;
+    _flushall();
+
+    // Exception happened, so we probably didn't clean up properly,
+    // Don't exit normally, just terminate
+    TerminateProcess(::GetCurrentProcess(), exceptionCode);
```
Comment (Yongqu) [inline bin/ch/ch.cpp]: when debugger not present?

File: lib/Runtime/Library/JavascriptFunction.cpp [hunk 2]
```diff
@@ -621,21 +619,38 @@ namespace Js
         }
 #endif
 
+        // mark volatile, because otherwise VC will incorrectly optimize away load in the finally block
+        volatile ulong exceptionCode = 0;
+        volatile int exceptionAction = EXCEPTION_CONTINUE_SEARCH;
+        EXCEPTION_POINTERS exceptionInfo = EXCEPTION_POINTERS();
```
Comment (Yongqu) [inline lib/Runtime/Library/JavascriptFunction.cpp]: is this basically {0}?

**Comment by Yongqu** [comment]
:+1:

File: bin/ch/ch.cpp (hunk 1 above)
Comment (MikeHolman) [inline bin/ch/ch.cpp]: I am trying to keep with current behavior, which had this in the except handler before.

**Pull request merged**